plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/importapp/ImportApplicationWizard.java

```diff
@@ -47,8 +50,28 @@ public ImportApplicationWizard() {
 		this.model = new ImportApplicationWizardModel();
 	}
 
+	public ImportApplicationWizard(Map<IProject, Collection<IBuildConfig>> projectsAndBuildConfigs) {
+		this();
+		if (projectsAndBuildConfigs != null && projectsAndBuildConfigs.size() == 1) {
+			Map.Entry<IProject, Collection<IBuildConfig>> entry = projectsAndBuildConfigs.entrySet().iterator().next();
+			IProject project = entry.getKey();
+			Connection connection = ConnectionsRegistryUtil.safeGetConnectionFor(project);
```

Comment:
If that blows here, then I don't know how to gracefully handle that. It's not any different that what is done in init().
I'd rather keep it as is, for the sake of simplicity.